x-pack/plugins/security_solution/server/lib/detection_engine/signals/threat_mapping/create_threat_signals.ts

```diff
@@ -87,10 +88,10 @@ export const createThreatSignals = async ({
 
   logger.debug(`Total event count: ${eventCount}`);
 
-  if (eventCount === 0) {
-    logger.debug(buildRuleMessage('Indicator matching rule has completed'));
-    return results;
-  }
+  // if (eventCount === 0) {
+  //   logger.debug(buildRuleMessage('Indicator matching rule has completed'));
+  //   return results;
+  // }
```

Comment:
Ah yeah this was interesting. I meant to make a TODO here to come back to this.

Essentially the Elasticsearch count api does not support runtime field mappings and so the eventCount is 0 if your query only has runtime fields in it. I think I could update the logic to be if runtimeMappings is empty, do the eventCount function but if there are runtime fields in the query we will have to forego this logic unfortunately. Is there another way we could determine the event count without relying on that Elasticsearch count api?

Reply:
I see, we should be able to just use `singleSearchAfter` with `size: 0` in that case and pull the total off of the search response

Reply:
ah ok. I'll give that a shot 👍

Reply:
cc: @nkhristinin ^

Reply:
Hey, it also makes sense to use `track_total_hits:true`
because we need to know the accurate amount of all events because later we will use it like:
```
  if (eventCount < threatListCount) {
```